artifacts-guidance.md

```diff
@@ -1,6 +1,30 @@
 # Guidance for Artifacts Authors
 
-Content other than OCI container images MAY be packaged using the image manifest.
-When this is done, the `config.mediaType` value should not be a known OCI image config [media type](media-types.md).
-Historically, due to registry limitations, some tools have created non-OCI conformant artifacts using the `application/vnd.oci.image.config.v1+json` value for `config.mediaType` and values specific to the artifact in `layer[*].mediaType`.
+## Artifacts and Images
+
+This specification is primarily concerned with packaging two kinds of content: Artifacts and Images.
+Both are representing using a [manifest](manifest.md).
+Images are defined in this specification as conformant content with a conformant [config](config.md), processed according to [conversion.md](conversion.md) to derive a [runtime-spec][] configuration blob.
+Conversely, an Artifact is any other conformant content that **does not contain a config to be interpreted by a runtime-spec implementation using the conversion mechanism.
+
+## Creating an Artifact
+
+Content other than Images MAY be packaged using the [manifest]; this is otherwise known as an Artifact.
+When this is done, the `artifactType` should be set to a custom media type, or the `config.mediaType` should not be a known Image config [media type](media-types.md).
 Implementation details and examples are provided in the [image manifest specification](manifest.md#guidelines-for-artifact-usage).
+
+Note: Historically, due to registry limitations, some tools have created non-conformant Artifacts using the `application/vnd.oci.image.config.v1+json` value for `config.mediaType`.
+
+## Interacting with Artifacts
+
+Software following the process described in [conversion.md](conversion.md) to create a [runtime-spec][] configuration blob SHOULD ignore unknown Artifacts (as determined by the presence of a descriptor `artifactType`) when selecting content from an [index](image-index.md).
+It is possible that implementations may also be able to interpret known Artifact types; however that is outside the scope of this spec.
+
+Artifacts can be detected at runtime using by checking two keys:
+1. Is an `artifactType` present in the descriptor, or in the [manifest](manifest.md)?
+2. Is the `config.mediaType` of the manifest something other than a [known media type](media-types.md) for [config](config.md)?
+
+If either of these tests is true, then the content is an Artifact.
```

Comment:
My understanding is that the wasm folks want to use the artifact type. It's runnable by the wasm runtime and not something that is aligned with the runtime spec. 
How can we support them by enabling the usage of `artifactType` ?  @cpuguy83 @tianon what do you folks think?

Reply:
That is covered by the case of "runtime has special knowledge of how to handle an artifact."

For the WASM-with-existing-containerd case, they should not set the artifact type as they will just be a "regular" image with special layers and a WASM platform object.

Reply:
I’m probably at the risk of swimming against the current here but I honestly was hoping that the artifact type could be set by anyone and not limited to. 
if the runtime spec doesn’t want the artifact type set then it can mandate that but WASM runtimes don’t need to honor that.

Reply:
The issue is that I need a way to tell "not runnable" from "runnable" in containerd; I'd rather not have "an image" that is also "an artifact" at the same time as we get back to the `platform: none/none` pattern.

Reply:
cc @jsturtevant; this is meant to enable (through clarity of what runtimes likely SHOULD support) efforts to make "native WASM images" work with "existing runtimes;" I see no reason that something that is otherwise a conformant image (and just uses a `wasm` platform) needs to set `artifactType`.

This is also somewhat related to https://github.com/opencontainers/image-spec/pull/1141#discussion_r1371576029, which I still want to keep as a follow-up as the scope of this PR has grown quite a bit.

Reply:
It's certainly possible; as discussed on the last PR having the standard config mediaType with a custom artifactType is rather outside what most would consider idiomatic, however.

The `platform` field on the descriptor should be the same as the platform in the `config` blob, so the platform of the image should be accessible via the same mechanism as the top-level `artifactType`.

Additionally, I do think that an annotation with e.g. "WASM image standard version" or similar would be perfectly idiomatic if you need some additional metadata.

WDYT?